File: go-sdk/pkg/models/operations/listdatabasepromotionrequests.go
package operations

type ListDatabasePromotionRequestsPathParams struct {
	Name         string `pathParam:"style=simple,explode=false,name=name"`
	Organization string `pathParam:"style=simple,explode=false,name=organization"`
}

type ListDatabasePromotionRequestsQueryParams struct {
	Page    *float64 `queryParam:"style=form,explode=true,name=page"`
	PerPage *float64 `queryParam:"style=form,explode=true,name=per_page"`
}

type ListDatabasePromotionRequestsRequest struct {
	PathParams  ListDatabasePromotionRequestsPathParams
	QueryParams ListDatabasePromotionRequestsQueryParams
}

type ListDatabasePromotionRequestsResponse struct {
	ContentType                                           string
	StatusCode                                            int64
	ListDatabasePromotionRequests200ApplicationJSONObject map[string]map[string]interface{}
}

File: ts-sdk/src/sdk/models/operations/listdatabasepromotionrequests.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";


export class ListDatabasePromotionRequestsPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=name" })
  name: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=organization" })
  organization: string;
}

export class ListDatabasePromotionRequestsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=page" })
  page?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=per_page" })
  perPage?: number;
}

export class ListDatabasePromotionRequestsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: ListDatabasePromotionRequestsPathParams;

  @SpeakeasyMetadata()
  queryParams: ListDatabasePromotionRequestsQueryParams;
}

export class ListDatabasePromotionRequestsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  listDatabasePromotionRequests200ApplicationJSONObject?: Record<string, Record<string, any>>;
}